src/ctia/observable/core.clj

```diff
@@ -0,0 +1,139 @@
+(ns ctia.observable.core
+  (:require [ctia.store
+             :refer
+             [calculate-verdict
+              list-judgements-by-observable
+              list-records
+              list-sightings-by-observables]]
+            [ctim.domain.id :as id]
+            [ctia.schemas.core
+             :refer
+             [APIHandlerServices Observable Reference Verdict]]
+
+            [ctia.properties :as p]
+            [schema.core :as s]))
+
+(s/defn observable->sightings
+  ([observable identity-map services]
+   (observable->sightings observable identity-map {} services))
+  ([observable :- Observable
+    identity-map
+    params
+    {{:keys [get-store]} :StoreService}]
+   (-> (get-store :sighting)
+       (list-sightings-by-observables
+        [observable]
+        identity-map
+        params))))
+
+(s/defn observable->judgements
+  ([observable identity-map services]
+   (observable->judgements observable identity-map {} services))
+  ([observable :- Observable
+    identity-map
+    params
+    {{:keys [get-store]} :StoreService}]
+   (-> (get-store :judgement)
+       (list-judgements-by-observable
+        observable
+        identity-map
+        params))))
+
+(defn observable->sighting-ids
+  [observable identity-map services]
+  (let [http-show (p/get-http-show services)
+        sightings (:data (observable->sightings observable identity-map services))]
+    (map #(id/long-id
+           (id/short-id->id :sighting (:id %) http-show))
+         sightings)))
+
+(defn observable->judgement-ids
+  [observable identity-map services]
+  (let [http-show (p/get-http-show services)
+        judgements (:data (observable->judgements observable identity-map services))]
+    (map #(id/long-id
+           (id/short-id->id :judgement (:id %) http-show))
+         judgements)))
+
+(defn get-relationships
+  [filters
+   identity-map
+   {{:keys [get-store]} :StoreService}]
+  (-> (get-store :relationship)
+      (list-records
+       {:all-of filters}
+       identity-map
+       {})
+      :data))
+
+(s/defn related-entity-ids
+  [filters
+   edge-end :- (s/enum :source_ref :target_ref)
+   entity-type
+   identity-map
+   services]
+  (let [relationships (get-relationships filters identity-map services)]
+    (->> (map edge-end relationships)
+         (map #(id/long-id->id %))
+         (filter #(= (str entity-type) (:type %)))
+         (map #(id/long-id %))
+         set)))
+
+(s/defn get-target-ids
+  [entity-ids
+   relationship_type
+   entity-type
+   identity-map
+   services]
```

Comment:
Too many positional arguments. Up to you, of course, I'd wrapped them into a map.